### PR TITLE
UAR-892: Fix Code Smells

### PR DESCRIPTION
**Pull request by RobertBromfield**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-892

### Change description

Small changes to reduce the number of code smells, which can be found here:

https://code-analysis.platform.aws.chdev.org/dashboard?id=uk.gov.companieshouse%3Aoverseas-entities-api

### Note

The code smell on line 139 of `OverseasEntitySubmissionDtoValidator` which is commented out code has been kept on request as it will be used in another ticket

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java [hunk 1]
```diff
@@ -32,58 +35,69 @@ public TrustCorporateValidator(AddressDtoValidator addressDtoValidator) {
...
+        }
+
+        validateAddress(TrustCorporateDto.REGISTERED_OFFICE_ADDRESS_FIELD,
+                trustCorporateDto.getRegisteredOfficeAddress(), errors, loggingContext);
+
+        boolean isSameAddressFlagValid = validateServiceAddressSameAsPrincipalAddress(
+                trustCorporateDto.getServiceAddressSameAsPrincipalAddress(), errors,
+                loggingContext);
+
+        if (isSameAddressFlagValid && Boolean.FALSE
```
Comment (mgibson-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java:73]: Would `Boolean.FALSE.equals(trustCorporateDto.getServiceAddressSameAsPrincipalAddress())` be better as `!trustCorporateDto.getServiceAddressSameAsPrincipalAddress()`? If it returns (object) Boolean then ignore this comment!

Comment (RobertBromfield) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java:73]: Honestly, I'm not sure I just copied this into another method to reduce the cognitive complexity.

It does return an instance of Boolean which may be null though I think `validateServiceAddressSameAsPrincipalAddress` checks for that.

Either way I'd say it's probably best to leave it

**Review by mgibson-ch — APPROVED** [review]

**Review by bmcgreevyCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidator.java [hunk 2]
```diff
@@ -59,16 +61,13 @@ public Errors validate(List<BeneficialOwnerCorporateDto> beneficialOwnerCorporat
...
-            if (Objects.nonNull(beneficialOwnerCorporateDto.getBeneficialOwnerNatureOfControlTypes())) {
-                fields.addAll(beneficialOwnerCorporateDto.getBeneficialOwnerNatureOfControlTypes());
-            }
-            if (Objects.nonNull(beneficialOwnerCorporateDto.getNonLegalFirmMembersNatureOfControlTypes())) {
-                fields.addAll(beneficialOwnerCorporateDto.getNonLegalFirmMembersNatureOfControlTypes());
-            }
-            if (Objects.nonNull(beneficialOwnerCorporateDto.getTrusteesNatureOfControlTypes())) {
-                fields.addAll(beneficialOwnerCorporateDto.getTrusteesNatureOfControlTypes());
-            }
+            var fields = Stream.of(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidator.java:64]: Do we have good unit test coverage for this? It looks good but want everything to be low risk.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java (hunk 1 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java:73]: I think if it's Boolean SonarQ wants Boolean.False.equals so I'd leave as is.

**Review by AndreKramer7654 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidator.java (hunk 2 above)
Comment (RobertBromfield) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidator.java:64]: Just ran the tests with coverage and looks like we do 

image

Give me a moment and I'll look to see if it's good enough though

Comment (RobertBromfield) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/BeneficialOwnerCorporateValidator.java:64]: Yeah looks like they're quite heavily tested with 
- testErrorReportedWhenNoNatureOfControlValuesAreAllNull
- testErrorReportedWhenNoNatureOfControlValuesAreAllEmpty
- testNoErrorReportedWhenSomeNatureOfControlValuesAreAllEmptyAndNull
- testErrorReportedWhenIsOnPublicRegisterNameFieldIsNull

**Pull request merged**